threedi_modelchecker/checks/timeseries.py

```diff
@@ -28,6 +36,93 @@ def description(self):
         return f"{self.column_name} contains an empty timeseries; remove the {self.table.name} instance or provide valid timeseries."
 
 
+class TimeSeriesEqualTimestepsCheck(BaseCheck):
+    """
+    Check that the timesteps in all timeseries in a column are equal.
+
+    This checks the timesteps for all timeseries in a column against the timesteps in the first
+    timeseries in that column. Consequently, if the first timeseries is wrong, all the other timeseries
+    in that column will raise a warning.
+    This check does not compare timeseries between different columns; for that, FirstTimeSeriesEqualTimestepsCheck
+    is used.
+    """
+
+    def get_invalid(self, session):
+        invalid_timeseries = []
+
+        first_timeseries = None
+
+        for row in self.to_check(session).all():
+            timeseries = row.timeseries
+
+            if not timeseries:
+                continue
+
+            if not first_timeseries:
+                first_timeseries = timeseries
+                continue  # don't compare first timeseries with itself
+
+            if not compare_timesteps(
+                first_timeseries=first_timeseries,
+                second_timeseries=timeseries,
+            ):
+                invalid_timeseries.append(row)
+
+        return invalid_timeseries
+
+    def description(self):
+        return (
+            f"One or more timesteps in {self.column_name} did not match the timesteps in the first timeseries in the column."
+            + "The timesteps in all timeseries must be the same."
+        )
+
+
+class FirstTimeSeriesEqualTimestepsCheck(BaseCheck):
+    """
+    Check that the timesteps in the first timeseries in the boundary condition columns are equal, if they both exist.
+
+    This is used in conjunction with TimeSeriesEqualTimestepsCheck to confirm that the timeseries in all the boundary
+    conditions have the same timesteps. If each timeseries within each column has the same timesteps, and the first
+    timeseries of each column matches timesteps with the first timeseries of each other column, then the timesteps of
+    all the timeseries must be the same.
+    """
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(column=models.BoundaryCondition1D.timeseries, *args, **kwargs)
+
+    def get_invalid(self, session):
+        invalid_timeseries = []
+
+        first_1d_timeseries = (
+            session.query(models.BoundaryCondition1D.timeseries.table)
+            .order_by(models.BoundaryCondition1D.id)
+            .limit(1)
+            .all()
+        )
+
+        first_2d_timeseries = (
+            session.query(models.BoundaryConditions2D.timeseries.table)
+            .order_by(models.BoundaryConditions2D.id)
+            .limit(1)
+            .all()
+        )
+
+        if first_1d_timeseries and first_2d_timeseries:
+            if not compare_timesteps(
+                first_timeseries=first_1d_timeseries[0][2],
```

Comment:
that `[2]` is assuming that hte `timeseries` is the third field, which may change in the future. I think `[0].timeseries` is also possible, because sqlalchemy returns `NamedTuples`.